### DISCUSSION TITLE
Agenda for the Flatcar community call on Tuesday, 8th of February 5:30pm CET / 10pm IST / 4:30pm GMT / 11:30am EST / 8:30am PST

### DISCUSSION THREAD

**Discussion by miao0miao** [flatcar/Flatcar#612] [Flatcar Office Hours]
# Agenda for the Flatcar community call on Tuesday,  8th of February 5:30pm CET / 10pm IST / 4:30pm GMT / 11:30am EST / 8:30am PST

- Zoom link: [https://us06web.zoom.us/j/84336116610](https://us06web.zoom.us/j/84336116610)
  - Meeting ID: 843 3611 6610
  - Passcode: 444888
- Youtube live stream / recording: [YouTube](https://www.youtube.com/watch?v=qke52eTZJQs&ab_channel=Kinvolk)
- Community Calls calendar (all future calls): [link](https://calendar.google.com/calendar/u/0/embed?src=c_ii991mqrpta9en8o7ofd4v19g4@group.calendar.google.com)
  - iCal version: [link](https://calendar.google.com/calendar/ical/c_ii991mqrpta9en8o7ofd4v19g4%40group.calendar.google.com/public/basic.ics)
- other ways to keep in touch [here](https://github.com/flatcar-linux/Flatcar#participate-and-contribute)

## Attendees
Mark Peek - VMware 
Tony Gosselin - Adobe
Joseph Sandoval -Adobe
Anuruddha Basak 
Kai Lüke 
Mathieu Tortuyaux 
William Light
Sayan Chowdhury
Dongsu Park
Danielle Tal 

## Welcome
- Brief intro / check-in of all participants in the Zoom call. Please introduce yourself and share what brings you here today.

## News
- first bug-smashing day of 2022!
  - ~40 bugs closed / fixed, see below for details
  - We use "good first issue" label for entry-level issues
  - There will be a new bug smashing day end of February, stay tuned! 
- Community / issues stats for January/February
  - See below
- Cluster API v1.1.0 released with Ignition / Flatcar support: https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.1.0
  - Ignition PR took ~10 months of work & review
  - Individual providers require separate PRs
  - Work in progress / ready to merge are Azure and AWS; VMware PR is almost feature-complete (we also take care of broader concerns like bump to v1.1.0) 

## Spotlight
- **Switch to CGroupsV2: Feedback, impact, and path forward**
  - November flatcar stable release made Cgroups V2 the default for new provisionings. 
  - We felt it was the right time to switch to V2, following mainstream V2 support in Docker and Kubernetes, as well as several other distributions having already switched to using V2 . 
  - Since we did not want to break existing workloads, all instances upgrading to that new major release would remain on Cgroups V1 
  - Something we did not account for is use cases and scenarios where users upgrade to new versions through re-provisioning, i.e. recycling the instance. 
  - Also, while we know many users do pull Beta updates, we over-estimated the use of the Beta channel releases across all scenarios. 
  - We received reports from a number of users who upgrade through instance recycling and are currently researching the best path forward. 
  - It’s already possible to configure Flatcar to use V1, but this currently requires interacting with the instance after provisioning, and also rebooting the instance once, or using a custom image. 
  - We are considering supporting an early boot time flag which can be set through provisioning, and would make the OS use Cgroups V1 from the start. Since systemd is already running  (and using cgroups) at the point where ignition data is being evaluated this requires thorough inspection and testing. 
  - We are also considering alternative approaches like shipping images which use CGroups v1 by default. 
  - In any case, we will provide a solution that will enable users to provision V1 instances. 
  - We expect to have it included in the late February release (the one after the ongoing release). 
  - **Learnings** 
  - For the Flatcar team, the lesson is that we need to consider all the scenarios in which a user will get a new version. 
  - For users, we continue to encourage everyone to run some instances with beta and to report any issues that you encounter, as we cannot test in all environments. 
We will halt promotion of beta to stable if we hear from users that something doesn’t work for them – but we can only do so if we get that feedback. 
- **Discussion**
  - Joseph shares his experience with the upstream kubernetes community re: cgroups v2 support. Many things aren't fully functioning with v2 - even 1.24 needs more validation - and upstream acknowledges this. SIG Node does not have a clear roadmap for full v2 support. This should be important for any kubernetes workload that uses Flatcar.
  - Kai shares features included in the upcoming support for Ignition v3 in Flatcar, which will among other things add options to set kernel parameters. Even if sometimes a reboot is required to make a kernel option apply, this reboot is transparent to provisioning.
  - Mathieu elaborates on his ignition v3 work and underlines that Flatcar will continue supporting v2 alongside v3.
  
- **FOSDEM wrap-up: Flatcar talks, other cool presentations**
  - How to make OS auto-updates boring: https://fosdem.org/2022/schedule/event/container_flatcar_autoupdate/
  - Extending the OS image (from sources): https://fosdem.org/2022/schedule/event/modding_the_immutable_how_to_extend_flatcar/
  - All's well that tests well: https://fosdem.org/2022/schedule/event/overview_of_flatcar_container_linux_test_framework/
  - Immutable Infrastructure with Flatcar Container Linux: https://fosdem.org/2022/schedule/event/flatcar_container_linux/
  - Securing and Hardening Container Host: https://fosdem.org/2022/schedule/event/securing_and_hardening_a_container_host/
- Question from youtube chat: When will talks be published?
  - Publishing has already started, the FOSDEM team work through all videos; all recordings should be available soon. Check the above links regularly, talks will pop up there.

## Release planning
- Ongoing and upcoming releases (https://github.com/orgs/flatcar-linux/projects/5)

## Q&A
- Questions from community participants, answered by the Flatcar maintainers

**Comment by miao0miao** [comment]
CAPI v1.1  - Add support for generating bootstrap data in Ignition format
https://github.com/kubernetes-sigs/cluster-api/pull/4172
https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.1.0

**Comment by miao0miao** [comment]
**Last month Issues opened/Closed (only Flatcar repo) up to February 7th:**

## Issues Opened
* [#619](https://github.com/flatcar-linux/Flatcar/issues/619):  alpha:3139.0.0
* [#618](https://github.com/flatcar-linux/Flatcar/issues/618):  Release Flatcar Container Linux Alpha 3139.0.0, Beta 3066.1.2, Stable 3033.2.2, LTS 2605.26.1
* [#617](https://github.com/flatcar-linux/Flatcar/issues/617):  [RFE] bpftool on Flatcar
* [#614](https://github.com/flatcar-linux/Flatcar/issues/614):  Improve text on Packer
* [#613](https://github.com/flatcar-linux/Flatcar/issues/613):  Community Call 2022-02-08
* [#611](https://github.com/flatcar-linux/Flatcar/issues/611):  After successful upgrade to 3033.2.0 the second reboot goes back to 2905.2.2
* [#609](https://github.com/flatcar-linux/Flatcar/issues/609):  Has the GCP image for 2765.2.6 been deleted or the required permissions changed?
* [#608](https://github.com/flatcar-linux/Flatcar/issues/608):  LSILogicParallel disk is still listed by lsblk after hot adding\removing
* [#607](https://github.com/flatcar-linux/Flatcar/issues/607):  [RFE] support for toolbox  without `docker.service` running
* [#606](https://github.com/flatcar-linux/Flatcar/issues/606):  CVE-2021-4034
* [#605](https://github.com/flatcar-linux/Flatcar/issues/605):  Slow ssh on gcp with oslogin
* [#604](https://github.com/flatcar-linux/Flatcar/issues/604):  Release Flatcar Container Linux Alpha 3127.0.0, Beta 3066.1.1, Stable 3033.2.1, LTS 2605.25.1
* [#603](https://github.com/flatcar-linux/Flatcar/issues/603):  Bug smashing Friday - 2022-01-28
* [#602](https://github.com/flatcar-linux/Flatcar/issues/602):  Handling CVE-2022-0185
* [#600](https://github.com/flatcar-linux/Flatcar/issues/600):  Release Team Call 2021-01-25
* [#599](https://github.com/flatcar-linux/Flatcar/issues/599):  motd: race in reboot coordinator report (probably only with flatcar.autologin)
* [#598](https://github.com/flatcar-linux/Flatcar/issues/598):  Can't install Kubernetes >=1.22 with RKE due to missing SELinux custom policies
* [#597](https://github.com/flatcar-linux/Flatcar/issues/597):  Performance degradation in 3033.2.0?
* [#596](https://github.com/flatcar-linux/Flatcar/issues/596):  Leaked selinux policy store files to the image root directory
* [#595](https://github.com/flatcar-linux/Flatcar/issues/595):  Release Flatcar Container Linux Alpha 3115.0.0

## Issues Closed
* [#619](https://github.com/flatcar-linux/Flatcar/issues/619):  alpha:3139.0.0
* [#611](https://github.com/flatcar-linux/Flatcar/issues/611):  After successful upgrade to 3033.2.0 the second reboot goes back to 2905.2.2
* [#604](https://github.com/flatcar-linux/Flatcar/issues/604):  Release Flatcar Container Linux Alpha 3127.0.0, Beta 3066.1.1, Stable 3033.2.1, LTS 2605.25.1
* [#603](https://github.com/flatcar-linux/Flatcar/issues/603):  Bug smashing Friday - 2022-01-28
* [#602](https://github.com/flatcar-linux/Flatcar/issues/602):  Handling CVE-2022-0185
* [#600](https://github.com/flatcar-linux/Flatcar/issues/600):  Release Team Call 2021-01-25
* [#596](https://github.com/flatcar-linux/Flatcar/issues/596):  Leaked selinux policy store files to the image root directory
* [#595](https://github.com/flatcar-linux/Flatcar/issues/595):  Release Flatcar Container Linux Alpha 3115.0.0
* [#589](https://github.com/flatcar-linux/Flatcar/issues/589):  Community Call 2022-01-11
* [#581](https://github.com/flatcar-linux/Flatcar/issues/581):  There is "metric 1024" for ip command
* [#564](https://github.com/flatcar-linux/Flatcar/issues/564):  Spliting Container Linux Configuration in multiple files
* [#561](https://github.com/flatcar-linux/Flatcar/issues/561):  `mantle`: rename `packet` to `equinix-metal`
* [#538](https://github.com/flatcar-linux/Flatcar/issues/538):  Unable to update from old version due to SSL certificate problem
* [#506](https://github.com/flatcar-linux/Flatcar/issues/506):  [cork] Failed to run create with UTF-8 user.name in git config
* [#500](https://github.com/flatcar-linux/Flatcar/issues/500):  [RFE] publish containerd versions via flatcar_production_image_packages.txt
* [#477](https://github.com/flatcar-linux/Flatcar/issues/477):  How to manualy set DNS during baremetal install (No DHCP)
* [#456](https://github.com/flatcar-linux/Flatcar/issues/456):  AMI for 2765.2.6 on AWS lost
* [#441](https://github.com/flatcar-linux/Flatcar/issues/441):  Simplify SDK format and distribution
* [#440](https://github.com/flatcar-linux/Flatcar/issues/440):  Simplify the SDK sources
* [#422](https://github.com/flatcar-linux/Flatcar/issues/422):  [RFE] Stricter dm-verity error handling
* [#397](https://github.com/flatcar-linux/Flatcar/issues/397):  DHCPv6 broken since 2765.2.0/Systemd 247
* [#394](https://github.com/flatcar-linux/Flatcar/issues/394):  Update Iptables from 1.6.2 to 1.8.x in flatcar OS version 2605.8.0
* [#384](https://github.com/flatcar-linux/Flatcar/issues/384):  IPv6 address auto-configuration doesn't work on AWS EC2 instances
* [#365](https://github.com/flatcar-linux/Flatcar/issues/365):  Include Amazon Root CAs in Flatcar's trust store
* [#340](https://github.com/flatcar-linux/Flatcar/issues/340):  flatcar image with open-vm-tools + cloud-init metadata from openstacks' metadata_urls
* [#331](https://github.com/flatcar-linux/Flatcar/issues/331):  [RFC] GPIO support
* [#328](https://github.com/flatcar-linux/Flatcar/issues/328):  flatcar 2605.9.0+ with systemd 246 journal logs produced is not readable by older versions
* [#291](https://github.com/flatcar-linux/Flatcar/issues/291):  [Tracking ] Update dev-vcs/repo to 2.8 in stable
* [#268](https://github.com/flatcar-linux/Flatcar/issues/268):  Remove unused repositories from manifest
* [#218](https://github.com/flatcar-linux/Flatcar/issues/218):  Stable 2605.6.0 systemd log spam Failed to set up mount unit: Invalid argument
* [#209](https://github.com/flatcar-linux/Flatcar/issues/209):  Add favicon.ico to Flatcar website
* [#204](https://github.com/flatcar-linux/Flatcar/issues/204):  Relabeling of content in /usr
* [#198](https://github.com/flatcar-linux/Flatcar/issues/198):  Make BPF tracing tools (bcc, bpftrace) easier to use in Flatcar
* [#187](https://github.com/flatcar-linux/Flatcar/issues/187):  UEFI Boot Support for Bare Metal Installs
* [#259](https://github.com/flatcar-linux/Flatcar/issues/259):  Docs does not mention what `UPDATE_STATUS_UPDATED_NEED_REBOOT` means
* [#178](https://github.com/flatcar-linux/Flatcar/issues/178):  [RFC] additional Reboot strategies needed?
* [#166](https://github.com/flatcar-linux/Flatcar/issues/166):  AWS C5A support
* [#270](https://github.com/flatcar-linux/Flatcar/issues/270):  Update ansible-bootstrap to Python 3
* [#243](https://github.com/flatcar-linux/Flatcar/issues/243):  [ignition] Add luks support
* [#135](https://github.com/flatcar-linux/Flatcar/issues/135):  Ignition ignores non-transpiled config without warnings or errors
* [#133](https://github.com/flatcar-linux/Flatcar/issues/133):  Swap file instructions don't work
* [#257](https://github.com/flatcar-linux/Flatcar/issues/257):  were these "Flatcar" images in GovCloud actually pushed by Flatcar?
* [#97](https://github.com/flatcar-linux/Flatcar/issues/97):  Ship Beta, Stable images for ARM
* [#95](https://github.com/flatcar-linux/Flatcar/issues/95):  Public feature / patch tracking system, automated Release Notes generation from PRs
* [#89](https://github.com/flatcar-linux/Flatcar/issues/89):  Run automatic tests for BPF before each Flatcar release
* [#88](https://github.com/flatcar-linux/Flatcar/issues/88):  Document dependencies of flatcar-install script
* [#43](https://github.com/flatcar-linux/Flatcar/issues/43):  ARM 32-bit builds on the download page
* [#38](https://github.com/flatcar-linux/Flatcar/issues/38):  Document how to build a custom image
* [#37](https://github.com/flatcar-linux/Flatcar/issues/37):  Can't upload 2345.3.0 as a custom image on Digital Ocean

**Comment by pothos** [comment]
Want to share it here, we have a script to generate cgroupv1 AWS EC2 images for custom AMIs now:
https://github.com/flatcar-linux/flatcar-docs/blob/main/create_cgroupv1_ami.sh
